.gitlab-ci.yml

```diff
@@ -0,0 +1,62 @@
+stages:
+  - deploy_to_sonatype
+  - create_key
+
+variables:
+  REGISTRY: 486234852809.dkr.ecr.us-east-1.amazonaws.com
+
+deploy_to_sonatype:
+  stage: deploy_to_sonatype
+  tags: [ "runner:docker", "size:large" ]
+  image: maven:3.8.2-jdk-8-slim
+  when: manual
+
+  script:
+    # Ensure we don't print commands being run to the logs during credential
+    # operations
+    - set +x
+
+    - apt update
+    - apt install -y python3 python3-pip
+    - python3 -m pip install awscli
+
+    - echo "Fetching Sonatype user..."
+    - export SONATYPE_USER=$(aws ssm get-parameter --region us-east-1 --name ci.jmeter-datadog-backend-listener.publishing.sonatype_username --with-decryption --query "Parameter.Value" --out text)
+
+    - echo "Fetching Sonatype password..."
+    - export SONATYPE_PASS=$(aws ssm get-parameter --region us-east-1 --name ci.jmeter-datadog-backend-listener.publishing.sonatype_password --with-decryption --query "Parameter.Value" --out text)
+
+    - echo "Fetching signing key password..."
+    - export GPG_PASSPHRASE=$(aws ssm get-parameter --region us-east-1 --name ci.jmeter-datadog-backend-listener.signing.gpg_passphrase --with-decryption --query "Parameter.Value" --out text)
+
+    - echo "Fetching signing key..."
+    - GPG_KEY=$(aws ssm get-parameter --region us-east-1 --name ci.jmeter-datadog-backend-listener.signing.gpg_private_key --with-decryption --query "Parameter.Value" --out text)
+
+    - printf -- "$GPG_KEY" | gpg --import --batch
```

Comment:
I would definitely consider safer ways to make sure we don't even accidentally print the GPG private key to the public job log by mistake.

Reply:
Another way is to save the key to a temporary file and use it for the gpg import. However, that method would still use `echo` or `printf` to output the contents of the env var into the file. It doesn't seem too different compared to doing it the current way. WDYT?

Reply:
Probably not, but use techniques to make sure that you cannot actually print it by mistake. For example, use a Bash block to capture a group of commands whose stderr/stdout you will then pipe to /dev/null.